### PR TITLE
remove blocking unsafeRunSync inside IO

### PR DESCRIPTION
**Pull request by yanns**

### CODE REVIEW

**Review by hamnis — APPROVED** [review]

File: tests/src/test/scala/org/http4s/EntityDecoderSuite.scala [hunk 1]
```diff
@@ -330,17 +330,16 @@ class EntityDecoderSuite extends Http4sSuite {
...
-    IO.async_[String] { cb =>
+    IO.async[String] { cb =>
       request
         .decodeWith(happyDecoder, strict = false) { s =>
           cb(Right(s))
           IO.pure(Response())
         }
-        .unsafeRunSync()
-      ()
+        .as(Some(IO.unit))
```
Comment (m-sp) [inline tests/src/test/scala/org/http4s/EntityDecoderSuite.scala:342]: shouldn't this return `None`? `async` has an optional cancelation token, and this is not really canceling anything right?

Comment (yanns) [reply tests/src/test/scala/org/http4s/EntityDecoderSuite.scala:342]: 👍 
yes you're right.

Comment (yanns) [reply tests/src/test/scala/org/http4s/EntityDecoderSuite.scala]: https://github.com/http4s/http4s/pull/4194/commits/53e61553a68eb68aff759388503de3ba9459344f

**Review by m-sp — APPROVED** [review]

**Comment by rossabaker** [comment]
The failing test is #4213.

**Pull request merged**